pkg/volume/csi/csi_driver_reg.go

```diff
@@ -0,0 +1,277 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package csi
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+	"sync"
+
+	"github.com/golang/glog"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+	corev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/util/retry"
+)
+
+const (
+	// Name of node annotation that contains JSON map of driver names to node
+	// names
+	annotationKey = "csi.volume.kubernetes.io/nodeid"
+)
+
+type csiDriver struct {
+	driverName     string
+	driverEndpoint string
+}
+
+// csiDrivers map keep track of all registered CSI drivers on the node and their
+// corresponding sockets
+var csiDrivers sync.Map
+
+// labelManagement is struct of channels used for communication between the driver registration
+// code and the go routine responsible for managing the node's labels
+type labelManager struct {
+	nodeName types.NodeName
+	k8s      kubernetes.Interface
+}
+
+var lm labelManager
```

Comment:
Here is what I was thinking. Optionally, put labelManager in to a new file, then do something like this:

```GO
package labelManager

import (
	"encoding/json"
	"fmt"
	"strings"
	"sync"

	"github.com/golang/glog"
	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
	"k8s.io/apimachinery/pkg/types"
	"k8s.io/client-go/kubernetes"
	corev1 "k8s.io/client-go/kubernetes/typed/core/v1"
	"k8s.io/client-go/util/retry"
)

type LabelManagerInterface interface {
	AddLabels(driverName string, ...)
}

type labelManagerStruct struct {
	nodeName types.NodeName
	k8s      kubernetes.Interface
}

func NewLabelManager(nodeName, kubeClient) LabelManagerInterface {
	return labelManagerStruct{
		nodeName: nodename,
		k8s: kubeClient,
	}
}

func (lm *labelManagerStruct)AddLabels(driverName string, ...) {
}
```

And in `func (p *csiPlugin) Init(host volume.VolumeHost) error` do this
```GO
    lm = labelManager.NewLabelManager(...)
```

And then in `RegistrationCallback(...)` do this:

```GO
lm.AddLabel(...)
```